### PR TITLE
Fixing some bugs in LWM2M

### PR DESCRIPTION
**Pull request by cabelitos**

### CODE REVIEW

File: src/lib/comms/sol-lwm2m.c [hunk 1]
```diff
@@ -307,6 +307,8 @@ static int bootstrap_with_server(struct sol_lwm2m_client *client,
 static int handle_resource(void *data, struct sol_coap_server *server,
     const struct sol_coap_resource *resource, struct sol_coap_packet *req,
     const struct sol_network_link_addr *cliaddr);
+static void remove_observer_entry(struct sol_ptr_vector *entries,
+    struct observer_entry *entry);
```
Comment (bsmelo) [inline src/lib/comms/sol-lwm2m.c]: Not needed.

**Comment by bsmelo** [comment]
Besides that prototype, everything lgtm.

Tested locally running samples and test-lwm2m as well.

**Comment by bsmelo** [comment]
Oh,

While your're at it, could you please remove ` * - Bootstrap.` from the list of unsupported features in `sol-lwm2m.h`?

File: src/lib/comms/sol-lwm2m.c (hunk 1 above)
Comment (cabelitos) [inline src/lib/comms/sol-lwm2m.c]: Indeed, I forgot to remove it. Ty

**Comment by bdilly** [comment]
:+1:

I would suggest you add fixed compiler warnings on commit msg next time.

**Comment by cabelitos** [comment]
merged

**Pull request closed without merging**